pkg/block/fetcher.go

```diff
@@ -49,10 +49,11 @@ const (
 	failedMeta    = "failed"
 
 	// Filter's label values.
-	labelExcludedMeta = "label-excluded"
-	timeExcludedMeta  = "time-excluded"
-	tooFreshMeta      = "too-fresh"
-	duplicateMeta     = "duplicate"
+	labelExcludedMeta     = "label-excluded"
+	timeExcludedMeta      = "time-excluded"
+	tooFreshMeta          = "too-fresh"
+	duplicateMeta         = "duplicate"
+	markedForDeletionMeta = "marked-for-deletion"
```

Comment:
Worth to comment that some blocks can be still loaded and not marked as `marked-for-deletion` because of some delays on purpose? (:

Reply:
I don't think this explanation is required. This would be required in case of deleting the block since we are not straight away deleting the block. Here we are marking the block straight away for deletion, so I don't think this is required. What do you think?

Reply:
I have added a detailed comment regarding this now.